### PR TITLE
DP-19337 Open/close behaviors in the hamburger menu

### PR DESCRIPTION
**Pull request by ygannett**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)

## Description

- Only one menu is open at a time including the "log in to" content.
     - Close pre open element as another is clicked to open.
     - Make the open/close animation consistent. 
- Adjust the position of the bottom border of "Log In To…" content in the hamburger menu.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-19337)

## Steps to Test

1. Go to http://localhost:3000/patterns/05-pages-Homepage-hamburger/05-pages-Homepage-hamburger.html.

2. Open the hamburger menu and click the menus randomly.

3. Find only one dropdown is open at a time. If any pre open element exist, it closes before another one opens.

4. Find the open/close animation is consistent whether open/close the same element or not.
     Note: this involves some illusions.  All menus has the same time setting for their duration. So, if the menu is long, the open/close movement looks faster than short one because the duration time is divided in their heights equally. As the menu gets longer, the assigned time to the same space would be less than one for shorter one.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

*

#### @TODO

*

### CODE REVIEW

File: patternlab/styleguide/source/assets/js/modules/mainNavHamburger.js [hunk 1]
```diff
@@ -15,94 +15,87 @@ if (null !== menuButtonText) {
...
+    // Set these elements unfocusable to enable to set focus on the first focusable element in hamburger menu with next tabbing as the menu opens.
+    document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .goog-te-menu-value").setAttribute("tabindex", "-1");
+    document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .direct-link").setAttribute("tabindex", "-1");
+    document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .js-util-nav-toggle").setAttribute("tabindex", "-1");
+  }
+});
 
-  //     if (menuButton.getAttribute("aria-expanded") === "false") {
-  //       // openMenu();
+// menuButton.addEventListener("keydown", function (e) {
```
Comment (tom-fleming) [inline patternlab/styleguide/source/assets/js/modules/mainNavHamburger.js]: @ygannett Can this commented code be removed if no longer needed?

**Comment by tom-fleming** [comment]
@ygannett Just a couple thing on this one. Can we remove unneeded commented code and also fix the conflicting code? Otherwise, LGTM!

**Comment by ygannett** [comment]
@tom-fleming It's cleaned up.  Can you take a look again?

File: patternlab/styleguide/source/assets/js/modules/utilNav.js [hunk 2]
```diff
@@ -5,7 +5,9 @@ export default (function (window, document, $, undefined) {
   let $utilityButtons = $('.js-util-nav-toggle');
 
   $panels.each(function () {
-    $panel = $(this);
+    if ($(this).closeset(".ma__header__hamburger__utility-nav--narrow ") !== true) {
```
Comment (tom-fleming) [inline patternlab/styleguide/source/assets/js/modules/utilNav.js]:
```suggestion
    if ($(this).closest(".ma__header__hamburger__utility-nav--narrow ") !== true) {

```
@ygannett

**Comment by tom-fleming** [comment]
@ygannett I merged in the changes from the main DP-17200 branch. Could you double-check this branch, and also take a look at the two comments I left above?

**Pull request merged**